### PR TITLE
Add payment and production configuration to form design UI

### LINKED ISSUE #561: Add production selection (tab) and payment backend/options configuration to React CRUD page
Currently, only the vanilla django backend supports these.

In #539, support for payment backends was added, but this is not added yet to the React UI to build forms.

**Tasks**

- [x] Add a tab "Payment"
- [x] Add the payment backend selection as a field
- [x] Add the payment options form as a field, if relevant (see #554) for similar problem with the registration backends
- [x] Ensure merchant can be selected from drop down list
- [x] Add a tab "Product"
- [x] Add a product selection field.

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes #561

### CODE REVIEW

File: src/openforms/js/components/admin/form_design/PaymentFields.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+            {
+                hasOptionsForm
+                ? (
+                    <FormRow>
+                        <Field
+                            name="form.paymentBackendOptions"
+                            label={<FormattedMessage description="Payment backend options label" defaultMessage="Payment backend options" />}
+                        >
+                            <Form
```
Comment (SilviaAmAm) [inline src/openforms/js/components/admin/form_design/PaymentFields.js:50]: I find the combination of `Field` and the `Form` from rjsf causes the help text to render a bit strange:
Screenshot from 2021-08-17 10-12-42

File: src/openforms/js/components/admin/form_design/RegistrationFields.js [hunk 2]
```diff
@@ -46,14 +47,14 @@ const RegistrationFields = ({
...
                 </Field>
             </FormRow>
-            {optionsFormSchema
+            {hasOptionsForm
                 ? (<FormRow>
                     <Field
                         name="form.registrationBackendOptions"
                         label={<FormattedMessage defaultMessage="Registration backend options" description="Registration backend options label" />}
                     >
                         <Form
```
Comment (SilviaAmAm) [inline src/openforms/js/components/admin/form_design/RegistrationFields.js:56]: Same remark as in the payment tab: the rendering of `Field` and `Form` seems a bit strange:
Screenshot from 2021-08-17 10-29-10

File: src/openforms/utils/mixins.py [hunk 3]
```diff
@@ -1,8 +1,37 @@
 from drf_jsonschema import to_jsonschema
 
+from openforms.api.utils import underscore_to_camel
+
+
+def _camelize_required(schema: dict):
+    """
+    Camelize the `required` field names, which are not picked up by drf camel case.
+    """
+    # import bpdb; bpdb.set_trace()
```
Comment (SilviaAmAm) [inline src/openforms/utils/mixins.py:10]:
```suggestion
```

**Review by SilviaAmAm — CHANGES_REQUESTED** [review]

File: src/openforms/js/compiled-lang/nl.json [hunk 4]
```diff
@@ -227,6 +227,12 @@
       "value": "Gebruik bestaande formulierdefinitie"
     }
   ],
+  "GV8HnZ": [
+    {
+      "type": 0,
+      "value": "Select payment backend"
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/compiled-lang/nl.json:233]: Yep, we'll go over them at the end of the sprint! Just simply ran `makemessages & compilemessages` to shut up the warnings in the console

File: src/openforms/js/components/admin/form_design/PaymentFields.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import Field from '../forms/Field';
+import FormRow from '../forms/FormRow';
+import Fieldset from '../forms/Fieldset';
+import Select from "../forms/Select";
+import Form from "@rjsf/core";
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/PaymentFields.js:9]: Caught by copy-paste... :see_no_evil:

File: src/openforms/js/components/admin/form_design/PaymentFields.js (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/PaymentFields.js:50]: It is - I looked if there was an easy CSS fix but there was not, and altering the layout didn't look straight forward either, so I decided to tackle that later.

**Pull request merged**